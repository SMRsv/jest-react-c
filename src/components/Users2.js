import React from 'react';

const Users2 = (props) => {
  const {user} = props;
  const legal = user.age >= 18 ? (
    <h2 style={{color: "green"}}>{user.name}</h2>
  ) : (
    <h2 style={{color: "red"}}>{user.name}</h2> 
  );
  return (
    <div data-testid={`user-${user.id}`}>{legal}</div>
  )
};

export default Users2;