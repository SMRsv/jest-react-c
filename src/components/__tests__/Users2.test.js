import {render, screen, cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";
import Users2 from "../Users2";

afterEach(()=>{
    cleanup();
});

test("Should render illegal user", () => {
    const user = {id: 1, name: "John", age: 15};
    render(<Users2 user={user} />);
    const userElement = screen.getByTestId("user-1");
    expect(userElement).toBeInTheDocument();
    expect(userElement).toHaveTextContent("John");
    expect(userElement).toContainHTML("orange");
});

test("Should render legal user", () => {
    const user = {id: 2, name: "Kelly", age: 20};
    render(<Users2 user={user} />);
    const userElement = screen.getByTestId("user-2");
    expect(userElement).toBeInTheDocument();
    expect(userElement).toHaveTextContent("Kelly");
    expect(userElement).toContainHTML("green");
});

test("Matches Snapshot", () => {
    const user = {id: 1, name: "John", age: 15};
    const tree = renderer.create(<Users2 user={user} />).toJSON();
    expect(tree).toMatchSnapshot();
});