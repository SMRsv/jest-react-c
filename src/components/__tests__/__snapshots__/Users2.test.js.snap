// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  data-testid="user-1"
>
  <h2
    style={
      Object {
        "color": "orange",
      }
    }
  >
    John
  </h2>
</div>
`;
